storage/scanner.go

```diff
@@ -125,28 +125,35 @@ func (rs *rangeScanner) RemoveRange(rng *Range) {
 	rs.removed <- rng
 }
 
+// heartBeat returns a duration between iterations to allow us to pace
+// the scan.
+func (rs *rangeScanner) heartBeat(start time.Time) time.Duration {
+	elapsed := time.Now().Sub(start)
+	remainingNanos := rs.interval.Nanoseconds() - elapsed.Nanoseconds()
+	if remainingNanos < 0 {
+		remainingNanos = 0
+	}
+	count := rs.iter.EstimatedCount()
+	if count < 1 {
+		count = 1
+	}
+	timeInterval := time.Duration(remainingNanos / int64(count))
+	log.V(6).Infof("next range scan iteration in %s", timeInterval)
+	return timeInterval
+}
```

Comment:
Yeah. It's easy to mess up pacing like this and doing so often doesn't show up easily.